### PR TITLE
Remove unneeded depends

### PR DESCRIPTION
**Pull request by weiwangncar**
TYPE: bug fix, clean up

KEYWORDS: dependencies, unused

SOURCE: internal

DESCRIPTION OF CHANGES:
Problem:
Some dependency is declared twice, some are no longer needed.

Solution:
Remove un-used depends.

LIST OF MODIFIED FILES: 
M      main/depend.common

TESTS CONDUCTED: 
1. Tested on Derecho with 6 proc to compile
2. The Jenkins tests are all passing.

RELEASE NOTE:

### CODE REVIEW

File: main/depend.common [hunk 1]
```diff
@@ -1417,10 +1415,6 @@ module_fdda_spnudging.o: \
 	../frame/module_wrf_error.o 
 
 
-module_sf_bep.o: \
```
Comment (islas) [inline main/depend.common:1420]: I believe this is still needed, from `phys/module_sf_bep.F`:
```
MODULE module_sf_bep
#if defined(mpas)
use mpas_atmphys_utilities, only: physics_error_fatal
#define FATAL_ERROR(M) call physics_error_fatal( M )
#else
use module_wrf_error
#define FATAL_ERROR(M) call wrf_error_fatal( M)
!USE module_model_constants
#endif
 USE module_sf_urban
 USE module_bep_bem_helper, ONLY: nurbm
```

**Review by islas — APPROVED** [review]
Disregard previous comment, I saw after that the removed dep was a duplicate

**Comment by weiwangncar** [comment]
The regression test results:

    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

File: main/depend.common [hunk 2]
```diff
@@ -1417,10 +1415,6 @@ module_fdda_spnudging.o: \
...
-	module_sf_urban.o 
-
-
```
Comment (dudhia) [inline main/depend.common:1423]: can you explain why this one is removed?

Comment (weiwangncar) [reply main/depend.common:1423]: @dudhia This dependency is declared elsewhere in this file, hence redundant.

**Review by dudhia — APPROVED** [review]

**Pull request merged**